File: leetcode/easy/Array/977_Squares of a Sorted Array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        
        for(int i = 0; i < nums.length; i++){
            if(nums[i] < 0){
                nums[i] = -nums[i];
            }
            nums[i] = nums[i] * nums[i];
        }
        
        Arrays.sort(nums);
        return nums;
        
    }
}

File: leetcode/easy/Array/cpp/977_Squares of a Sorted Array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
    
        vector<int> ans(nums.size());
        
        int l = 0, r = nums.size()-1;
        
        for(int i = nums.size()-1; i >= 0; i--){
            
            if(abs(nums[r]) > abs(nums[l])){
                ans[i] = nums[r] * nums[r];
                r--;
            } 
            else{
                ans[i] = nums[l] * nums[l];
                l++;
            } 
                
        }
        
        return ans;
    }
};